LiveLines/ClientApp/src/pages/Home.js

```diff
@@ -8,44 +8,78 @@ import { BiMusic } from 'react-icons/bi';
 export function Home() {
   const [line, setLine] = useState("");
   const [lines, setLines] = useState([]);
+  const [linePlaceholder, setLinePlaceholder] = useState("");
   
   const [songEnabled, setSongEnabled] = useState(false);
   const [songInput, setSongInput] = useState('');
-  
+
+  const [canPostToday, setCanPostToday] = useState(null);
+  const [canPostYesterday, setCanPostYesterday] = useState(null);
+  const [postYesterdayInput, setPostYesterdayInput] = useState(false);
+
   const { user, loginAttempted } = useContext(UserContext);
 
   // TODO: Move this kind of logic to a service layer .js file
-  const parseLine = line => ({ ...line, createdAt: parseISO(line.createdAt) });
+  const parseLine = line => ({ ...line, dateFor: parseISO(line.dateFor) });
 
-  useEffect(() => {
-    const getLines = async () => {
-      const resp = await getData("api/lines");
-      const linesResp = await resp.json();
-      setLines(linesResp.map(parseLine));
-    }
+  const getLines = async () => {
+    const resp = await getData("api/lines");
+    const linesResp = await resp.json();
+    setLines(linesResp.map(parseLine));
+  }
 
+  const getLineOperations = async () => {
+    const resp = await getData("api/lineOperations");
+    const operationsResp = await resp.json();
+    setCanPostToday(operationsResp.canPostToday);
+    setCanPostYesterday(operationsResp.canPostYesterday);
+    if (!operationsResp.canPostYesterday) {
+      setPostYesterdayInput(false);
+    }
```

Comment:
Since `canPostYesterday` will be ran once per page load, and `postYesterdayInput` will start as false, do we need to set this to false?

Reply:
This is required to be run after a line is posted as well as on page load